src/d3d9/d3d9_device.cpp

```diff
@@ -3791,6 +3791,9 @@ namespace dxvk {
         m_dirtySamplerStates |= 1u << StateSampler;
       }
 
+      m_drefClamp &= ~(1u << StateSampler);
+      m_drefClamp |= newTexture->IsUpgradedToD32f() << StateSampler;
```

Comment:
Shifting a bool isn't going to work for samplers >= 8.

Just write this the same as the others.